core/core-workflow-steps/src/main/java/org/eclipse/sw360/antenna/workflow/analyzers/CsvAnalyzerImpl.java

```diff
@@ -139,15 +136,15 @@ private ArtifactCoordinates createCoordinates(CSVRecord record) {
 
         switch (type) {
             case "mvn":
-                return new MavenCoordinates(record.get(NAME), record.get(GROUP), record.get(VERSION));
```

Comment:
To be honest I hoped that we get rid of these switch stuff with the change. A constructor that takes type, namespace, name and version and perhaps one without namespace. This looks very complex and it actually always the same.

Reply:
the `ArtifactCoordinatesUtils` are just a tool to make the code migration easy. This is an intermediate state of the migration, of a WIP PR.

Reply:
With the current dev state, one would write:
```java
    private Coordinate createCoordinate(CSVRecord record) {
        final Coordinate.Builder builder = Coordinate.builder()
                .withName(record.get(NAME))
                .withVersion(record.get(VERSION));

        String type = record.isMapped(COORDINATE_TYPE) ? record.get(COORDINATE_TYPE) : "mvn";
        switch (type) {
            case "mvn":
                builder.withType(Coordinate.Types.MAVEN);
                builder.withNamespace(record.get(GROUP));
            case "dotnet":
                builder.withType(Coordinate.Types.NUGET);
            case "javascript":
                builder.withType(Coordinate.Types.NPM);
                builder.withNamespace(record.get(GROUP));
            case "bundle":
                builder.withType(Coordinate.Types.BUNDLE);
        }
        
        return builder.build();
    }
```

Reply:
RIght, but wouldn't be the default generic which would be the default case of the switch?